soda_machine.rb

```diff
@@ -7,12 +8,18 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+  args[:sodas].length
```

Comment:
`args` is only defined in the `SodaMachine::initialize` scope.  You don't have access to it here.  Instead, use the instance variable `@sodas`.